python/ray/tests/test_client_proxy.py

```diff
@@ -30,7 +31,8 @@ def test_proxy_manager_lifecycle(shutdown_only):
     os.environ["TIMEOUT_FOR_SPECIFIC_SERVER_S"] = "5"
     pm = proxier.ProxyManager(ray_instance["redis_address"],
                               ray_instance["session_dir"])
-    pm._free_ports = [45000, 45001]
+    port_one, port_two = random.choices(range(45000, 45100), k=2)
```

Comment:
```suggestion
	# NOTE: We use different ports between runs because sometimes the port is
    # not released, introducing flakiness.
    port_one, port_two = random.choices(range(45000, 45100), k=2)
```